### PR TITLE
Small improvements for Prelude.Natural

### PR DESCRIPTION
**Pull request by sjakobi**
* Add Natural.subtract
* Small improvements for the Natural comparison functions
  * Add property assertions
  * Only lessThanEqual uses the Natural/subtract primitive, everything
    else is ultimately implemented in terms of lessThanEqual.
  * greaterThanEqual requires only 2 normalization steps instead of 3

### CODE REVIEW

**Comment by sjakobi** [comment]
How can I update `nixops/dhall-haskell.json`?

**Comment by Gabriel439** [comment]
@sjakobi: Run:

```
$ nix-prefetch-git --fetch-submodule git@github.com:dhall-lang/dhall-haskell.git
```

... and if you need a specific revision then add it to the end of the command

File: Prelude/Natural/equal [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-      → λ(b : Natural)
-      →     Natural/isZero (Natural/subtract a b)
-        &&  Natural/isZero (Natural/subtract b a)
+    = λ(a : Natural) → λ(b : Natural) → lessThanEqual a b && lessThanEqual b a
 
 let example0 = assert : equal 5 5 ≡ True
 
 let example1 = assert : equal 5 6 ≡ False
 
+let property0 = λ(n : Natural) → assert : equal n n ≡ True
```
Comment (Gabriel439) [inline Prelude/Natural/equal]: Note that `dhall lint` currently drops `assert`s that are nested under a lambda.  This is something I've been meaning to fix, but until it is fixed CI will reject this

File: Prelude/Natural/equal [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+        ./lessThanEqual sha256:1a5caa2b80a42b9f58fff58e47ac0d9a9946d0b2d36c54034b8ddfe3cb0f3c99
+      ? ./lessThanEqual
+
 let equal
     : Natural → Natural → Bool
-    =   λ(a : Natural)
-      → λ(b : Natural)
-      →     Natural/isZero (Natural/subtract a b)
-        &&  Natural/isZero (Natural/subtract b a)
+    = λ(a : Natural) → λ(b : Natural) → lessThanEqual a b && lessThanEqual b a
```
Comment (singpolyma) [inline Prelude/Natural/equal]: Why is this better?

**Comment by sjakobi** [comment]
@singpolyma I guess it was just my personal preference to re-use a prelude function here. The normalized code is of course the same (modulo commutativity maybe).

**Pull request merged**